src/main/java/uk/gov/ons/ctp/response/casesvc/client/ActionSvcClient.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
+import uk.gov.ons.ctp.common.rest.RestUtility;
+import uk.gov.ons.ctp.response.action.representation.ActionDTO;
+import uk.gov.ons.ctp.response.action.representation.ActionPlanDTO;
+import uk.gov.ons.ctp.response.casesvc.config.AppConfig;
+
+/** The impl of the service which calls the action service via REST */
+@Service
+public class ActionSvcClient {
+  private static final Logger log = LoggerFactory.getLogger(ActionSvcClient.class);
+
+  @Autowired private AppConfig appConfig;
+
+  @Autowired private RestTemplate restTemplate;
+
+  @Qualifier("actionServiceRestUtility")
+  @Autowired
+  private RestUtility restUtility;
+
+  /**
+   * Make use of the ActionService to create and post a new Action for a given caseId according to
+   * Category actionType and CaseEvent createdBy values
+   *
+   * @param actionType action type
+   * @param caseId the UUID caseId
+   * @param createdBy who did this
+   */
+  public void createAndPostAction(String actionType, UUID caseId, String createdBy) {
+    ActionDTO actionDTO = new ActionDTO();
+    actionDTO.setCaseId(caseId);
+    actionDTO.setActionTypeName(actionType);
+    actionDTO.setCreatedBy(createdBy);
+    HttpEntity<ActionDTO> httpEntity = restUtility.createHttpEntity(actionDTO);
+
+    UriComponents uriComponents =
+        restUtility.createUriComponents(appConfig.getActionSvc().getActionsPath(), null);
+
+    log.with("case_id", caseId).debug("about to post to the Action SVC");
+    restTemplate.exchange(uriComponents.toUri(), HttpMethod.POST, httpEntity, ActionDTO.class);
+  }
+
+  /**
+   * Get actionplans from action service using action plan selectors
+   *
+   * @param collectionExerciseId Id of collection exercise to get action plans for
+   * @param activeEnrolments selector to determine if actionplan for active enrolments(email) or
+   *     not(letters)
+   * @return List of action plans matching selectors
+   */
+  public List<ActionPlanDTO> getActionPlans(UUID collectionExerciseId, boolean activeEnrolments) {
+    log.with("collection_exercise_id", collectionExerciseId)
+        .with("active_enrolments", activeEnrolments)
+        .debug("Retrieving action plan for selectors");
+
+    MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add("collectionExerciseId", collectionExerciseId.toString());
+    queryParams.add("activeEnrolment", Boolean.toString(activeEnrolments));
+    UriComponents uriComponents =
+        restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
+
+    ResponseEntity<List<ActionPlanDTO>> responseEntity;
+    HttpEntity<?> httpEntity = restUtility.createHttpEntityWithAuthHeader();
+    try {
+      responseEntity =
+          restTemplate.exchange(
```

Comment:
Not sure but this emplay that CSRF is disabled.

Reply:
CSRF is irrelevant behind the firewall